src/main/java/io/netty/incubator/codec/http3/QpackEncoder.java

```diff
@@ -92,36 +431,36 @@ private void encodeLiteral(ByteBuf out, CharSequence name, CharSequence value) {
      * <a href="https://tools.ietf.org/html/rfc7541#section-5.2">Section 5.2</a>.
      */
     private void encodeStringLiteral(ByteBuf out, CharSequence value) {
-        encodeStringLiteral(out, (byte) 0x80, 7, value);
+        //    0   1   2   3   4   5   6   7
+        // +---+---+---+---+---+---+---+---+
+        // | H |    String Length (7+)     |
+        // +---+---------------------------+
+        // |  String Data (Length octets)  |
+        // +-------------------------------+
+        // TODO: Force H = 1 till we support sensitivity detector
+        encodeLengthPrefixedHuffmanEncodedLiteral(out, (byte) 0b1000_0000, 7, value);
     }
 
     /**
-     * Encode string literal according to Section 5.2.
-     * <a href="https://tools.ietf.org/html/rfc7541#section-5.2">Section 5.2</a>.
+     * Encode a string literal.
      */
-    private void encodeStringLiteral(ByteBuf out, byte mask, int prefix, CharSequence value) {
+    private void encodeLengthPrefixedHuffmanEncodedLiteral(ByteBuf out, byte mask, int prefix, CharSequence value) {
         int huffmanLength = huffmanEncoder.getEncodedLength(value);
         encodePrefixedInteger(out, mask, prefix, huffmanLength);
         huffmanEncoder.encode(out, value);
     }
 
-    /**
-     * Encode integer according to
-     * <a href="https://tools.ietf.org/html/rfc7541#section-5.1">Section 5.1</a>.
-     */
-    private static void encodePrefixedInteger(ByteBuf out, byte mask, int prefixLength, int i) {
-        int nbits = (1 << prefixLength) - 1;
-        if (i < nbits) {
-            out.writeByte((byte) (mask | i));
-        } else {
-            out.writeByte((byte) (mask | nbits));
-            int remainder = i - nbits;
-            while (remainder > 128) {
-                byte next = (byte) ((remainder % 128) | 0x80);
-                out.writeByte(next);
-                remainder = remainder / 128;
-            }
-            out.writeByte((byte) remainder);
+    private boolean mayBlockStream() {
+        return blockedStreams < maxBlockedStreams - 1;
+    }
+
+    private static final class StreamTracker extends ArrayList<Integer> {
```

Comment:
nit: we could even be more fancy here and just roll our own data structure which uses `int[]`. This way we can get rid of boxing